qt/scientific_interfaces/Muon/ALCInterface.cpp

```diff
@@ -285,5 +286,116 @@ void ALCInterface::importPeakData(const std::string &workspaceName) {
+  switch (m_ui.stepView->currentIndex()) {
+  case DataLoading:
+    externalPlotDataLoading();
+    break;
+  case BaselineModel:
+    externalPlotBaselineModel();
+    break;
+  case PeakFitting:
+    externalPlotPeakFitting();
+    break;
+  }
+}
+
+/**
+ * Plots in workbench the single workspace from the data given
+ * @param data The workspace to add to the ADS before plotting
+ * @param workspaceName The names of workspace to plot
+ * @param workspaceIndices String list of indices to plot (e.g.
+ * '0-2,5,7-10')
+ * @param errorBars Boolean to add/remove error bars to plot
+ * @param kwargs The kwargs used when plotting the workspace
+ */
+void ALCInterface::externallyPlotWorkspace(MatrixWorkspace_sptr &data, std::string const &workspaceName,
+                                           std::string const &workspaceIndices, bool errorBars,
+                                           boost::optional<QHash<QString, QVariant>> const &kwargs) {
+  AnalysisDataService::Instance().addOrReplace(workspaceName, data);
+  m_externalPlotter->plotSpectra(workspaceName, workspaceIndices, errorBars, kwargs);
+}
+
+/**
+ * Plots in workbench all the provided workspaces from the data given
+ * @param data The workspace to add to the ADS before plotting
+ * @param workspaceNames List of names of workspaces to plot
+ * @param workspaceIndices List of indices to plot
+ * @param errorBars List of booleans to add/remove error bars to each line individually
+ * @param kwargs The kwargs used when plotting each of the workspaces
+ */
+void ALCInterface::externallyPlotWorkspaces(MatrixWorkspace_sptr &data, std::vector<std::string> const &workspaceNames,
+                                            std::vector<int> const &workspaceIndices,
+                                            std::vector<bool> const &errorBars,
+                                            std::vector<boost::optional<QHash<QString, QVariant>>> const &kwargs) {
+  AnalysisDataService::Instance().addOrReplace(workspaceNames[0], data);
+  m_externalPlotter->plotCorrespondingSpectra(workspaceNames, workspaceIndices, errorBars, kwargs);
+}
+
+/**
+ * Handle Data Loading external plot requested. Will plot the loaded data if available
+ */
+void ALCInterface::externalPlotDataLoading() {
+  auto data = m_dataLoading->exportWorkspace();
+  if (data) {
+    QHash<QString, QVariant> kwargs;
+    kwargs.insert("marker", QString(".").toLatin1().constData());
+    kwargs.insert("linestyle", QString("None").toLatin1().constData());
+    externallyPlotWorkspace(data, "ALC_External_Plot_Loaded_Data", "0", true, kwargs);
+  } else
+    logger.warning("Load some data before externally plotting");
+}
+
+/**
+ * Handle Baseline Model external plot requested. Will plot the baseline model data if available otherwise will plot the
+ * loaded data if available
+ */
+void ALCInterface::externalPlotBaselineModel() {
+  auto data = m_baselineModellingModel->exportWorkspace();
+  if (data) {
```

Comment:
We can do this in one line

```suggestion
  if (auto const data = m_baselineModellingModel->exportWorkspace()) {
```